gateway/app/endpoints/geofencing_subscriptions/create_subscription.py

```diff
@@ -0,0 +1,38 @@
+from http import HTTPStatus
+
+from fastapi import APIRouter
+from fastapi.exceptions import RequestValidationError
+
+from app.drivers.geofencing import GeofencingSubscriptionInterfaceDep
+from app.exceptions import ApiException
+from app.schemas.geofencing import Protocol, Subscription, SubscriptionRequest
+
+router = APIRouter()
+
+
+@router.post("/subscriptions")
+async def post_subscriptions(
+    req: SubscriptionRequest,
+    geofencing_subscription_interface: GeofencingSubscriptionInterfaceDep,
+) -> Subscription:
+    if req.protocol != Protocol.HTTP:
+        raise ApiException(
+            status=HTTPStatus.BAD_REQUEST,
+            code="INVALID_PROTOCOL",
+            message="Only HTTP is supported.",
+        )
+
+    device = req.config.subscriptionDetail.device
+    if device is None:
+        raise RequestValidationError("Device must be set")
```

Comment:
The correct error is 422 MISSING_IDENTIFIER, here's some code I've written for it:
```
    @app.exception_handler(MissingDevice)
    async def missing_identifier_found_exception_error(
        request: Request, exc: MissingDevice
    ) -> Response:
        body = ErrorInfo(
            status=422,
            code="MISSING_IDENTIFIER",
            message="The device cannot be identified.",
        )
        return JSONResponse(status_code=422, content=jsonable_encoder(body))
```